### PR TITLE
Backend login allow user to have not persistent connection

### PR DESCRIPTION
**Pull request by ChVuagniaux**
Currently all backend connection are persistent ( the lifetime of the cookie is 5 year ), this is perfect for many case. But for project that require more security this is not very good.

This PR add a config parameter `cms.backendLoginRemember` that allow administrators to choose the wanted behavior of this remember me feature.

value | description
-- | :---
true (by default) | like now : session not expire
false | session life time is the value of `session.lifetime` (2 hours by default)
null | display a checkbox on the login page that allow user to choose 

This PR can fix the following issue #2394

*This commit is inspired from the PR of @Nyorai ( https://github.com/Nyorai/october/pull/1)*

### CODE REVIEW

File: modules/backend/lang/en/lang.php [hunk 1]
```diff
@@ -39,6 +39,7 @@
         'restore' => 'Restore',
         'login_placeholder' => 'login',
         'password_placeholder' => 'password',
+        'remember_me' => 'Keep me logged in?',
```
Comment (LukeTowers) [inline modules/backend/lang/en/lang.php]: Change this string to "Stay logged in" please

File: config/cms.php [hunk 2]
```diff
@@ -329,4 +329,21 @@
...
+    | true  - session never expire (cookie expiration in 5 years)
+    |
+    | false - session have a limited time (see session.lifetime)
+    |
+    | null  - The form login display a checkbox that allow user to choose
+    |         wanted behavior
+    |
+    */
+
+    'backendLoginRemember' => true,
```
Comment (LukeTowers) [inline config/cms.php]: Please rename to `backendForceRemember`, and move this to just below https://github.com/octobercms/october/blob/master/config/cms.php#L52

**Comment by ChVuagniaux** [comment]
@LukeTowers I have applied requested changes

File: modules/backend/controllers/auth/signin.htm [hunk 3]
```diff
@@ -32,6 +32,21 @@
             </button>
         </div>
 
+        <?php if (is_null(config('cms.backendForceRemember', true))) : ?>
+        <!-- Remember checkbox -->
```
Comment (LukeTowers) [inline modules/backend/controllers/auth/signin.htm]: Could you please indent the contents of this if block?

**Comment by LukeTowers** [comment]
@ChVuagniaux as soon as you make the one other minor formatting change, I'm fine with merging this, assuming that @daftspunk doesn't have any other input on it.

Thanks for your work, it's a nice and simple solution to a problem that's been on my backburner for months now :)

**Comment by ChVuagniaux** [comment]
@LukeTowers change done

Happy to contribute one more time to this AWESOME CMS :)

What do you think of a similar approach for the [rainlab/user-plugin](https://github.com/rainlab/user-plugin) with a settings in backend configuration ?

**Comment by LukeTowers** [comment]
I like the idea, perhaps also enabling defaulting session length to the config value but being able to override it with a plugin setting?

**Pull request merged**